micromamba/src/run.cpp

```diff
@@ -165,13 +379,18 @@ set_run_command(CLI::App* subcom)
         }
 #endif
 
-        int status, pid;
+        int status;
+        PID pid;
         std::error_code ec;
 
         ec = proc.start(wrapped_command, opt);
 
         std::tie(pid, ec) = proc.pid();
 
+#ifndef _WIN32
+        static ScopedProcFile scoped_proc_file{ process_name, pid, std::move(proc_dir_lock) }; // Note: this object is static only so that calls to `std::exit()` will call the destructor.
```

Comment:
I think it might be simpler to only store the parent process ID? Everything may be much simpler & faster with locking etc.

The OS should have a way to find the child processes if we need & want to.

Reply:
But the process name is something we determine, right? I don't think I understand :)

We want to store `process_name` in a file that is called <pid>.json -- and then later we can map the pid's to process names.